packages/transaction-controller/src/TransactionController.ts

```diff
@@ -311,26 +533,31 @@ export class TransactionController extends BaseControllerV1<
       newTransactionMeta,
       'TransactionController#failTransaction - Add error message and set status to failed',
     );
-    this.onTransactionStatusChange(newTransactionMeta);
```

Comment:
It was indeed simple, but it was encapsulating the name of the event and the event payload construction to avoid duplicating that on every usage.

For the sake of review and audit, i'd again encourage the minimum possible changes here to ensure the least risk and to limit the scope of the PR.